selfdrive/ui/ui.cc

```diff
@@ -471,9 +426,7 @@ void handle_message(UIState *s, SubMaster &sm) {
 }
 
 static void check_messages(UIState *s) {
-  while (true) {
-    if (s->sm->update(0) == 0)
-      break;
+  if (s->sm->update(0) > 0){
```

Comment:
while

Reply:
sm->update() will set all updated to false before polling new message. if we use while here,we can't use sm->updated() out of handle_message again. (currently,sm.updated("model") and sm.updated("radarState") is used in paint.cc)

the sockets in SubMaster are always conflates, so maybe we don't have do use while here ?

Reply:
The while is just to preserve the old behavior, but it should be fine with the if.

Reply:
old version's SubSocket are not conflates. I think we can remove while now without any problems